lms/djangoapps/grades/models.py

```diff
@@ -47,11 +58,12 @@ def to_json(self):
         if self._json is None:
             sorted_blocks = sorted(self, key=attrgetter('locator'))
             list_of_block_dicts = [block._asdict() for block in sorted_blocks]
-            course_key = next(iter(self)).locator.course_key  # all blocks are from the same course
+            course_key_string = self._get_course_key_string() # all blocks are from the same course
+
             for block_dict in list_of_block_dicts:
                 block_dict['locator'] = unicode(block_dict['locator'])  # BlockUsageLocator is not json-serializable
             data = {
-                'course_key': unicode(course_key),
+                'course_key': course_key_string,
```

Comment:
Can you add a quick unit test for an empty `BlockRecordSet`?  These edge cases make me think it's worthwhile to have a local test for it.